core/rpc/src/rpc_impl/operation.rs

```diff
@@ -0,0 +1,343 @@
+                    .payload(),
+            );
+            let script_hash = blake2b_160(script.as_slice());
+            let key = add_prefix(
+                *SCRIPT_HASH_EXT_PREFIX,
+                script_hash::Key::ScriptHash(script_hash).into_vec(),
+            );
+
+            batch.put_kv(key, script_hash::Value::Script(&script))?;
+            ret.push(H160(script_hash));
+        }
+
+        batch.commit()?;
+
+        Ok(ret)
+    }
+
+    pub(crate) fn inner_get_generic_block(
+        &self,
+        txs: Vec<packed::Transaction>,
+        block_num: u64,
+        block_hash: H256,
+        parent_hash: H256,
+        timestamp: u64,
+    ) -> Result<GenericBlock> {
+        let mut res: Vec<GenericTransaction> = Vec::new();
+
+        for tx in txs.into_iter() {
+            let tx_hash = tx.calc_tx_hash();
+            res.push(
+                self.inner_get_generic_transaction(
+                    tx,
+                    tx_hash.unpack(),
+                    TransactionStatus::Committed,
+                )?
+                .into(),
+            );
+        }
+
+        Ok(GenericBlock::new(
+            block_num,
+            block_hash,
+            parent_hash,
+            timestamp,
+            res,
+        ))
+    }
+
+    pub(crate) fn inner_get_generic_transaction(
+        &self,
+        tx: packed::Transaction,
+        tx_hash: H256,
+        tx_status: TransactionStatus,
+    ) -> Result<GenericTransactionWithStatus> {
+        let mut id = 0;
+        let mut ops = Vec::new();
+        let tx_view = tx.into_view();
+
+        for input in tx_view.inputs().into_iter() {
+            let cell = self
+                .get_detailed_live_cell(&input.previous_output())?
+                .unwrap();
+            let mut op = self.build_operation(&mut id, &cell.cell_output, &cell.cell_data, true)?;
+            ops.append(&mut op);
+            id += 1;
+        }
+
+        for (cell, data) in tx_view.outputs_with_data_iter() {
+            let data = data.pack();
+            let mut op = self.build_operation(&mut id, &cell, &data, false)?;
+            ops.append(&mut op);
+            id += 1;
+        }
+
+        let generic_tx = GenericTransaction::new(tx_hash, ops);
+
+        Ok(GenericTransactionWithStatus::new(generic_tx, tx_status))
+    }
+
+    #[allow(clippy::if_same_then_else)]
+    pub(crate) fn build_operation(
+        &self,
+        id: &mut u32,
+        cell: &packed::CellOutput,
+        cell_data: &packed::Bytes,
+        is_input: bool,
+    ) -> Result<Vec<Operation>> {
+        let mut ret = Vec::new();
+        let normal_address = Address::new(self.net_ty, cell.lock().into());
+
+        if self.is_sudt(&cell.type_()) {
+            let udt_amount = InnerAmount {
+                value: self.get_udt_amount(is_input, cell_data.raw_data()),
+                udt_hash: cell.type_().to_opt().map(|s| s.calc_script_hash().unpack()),
+                status: Status::Unconstrained,
```

Comment:
for cheque, status should be fleeting

Reply:
for other script, udt status should be locked

Reply:
Why the udt status is locked except cheque?

Reply:
because when collecting udt inputs, mercury will not use other script, so these udt behave like locked